OrbitGl/SamplingReport.h

```diff
@@ -5,50 +5,55 @@
 #ifndef ORBIT_GL_SAMPLING_REPORT_H_
 #define ORBIT_GL_SAMPLING_REPORT_H_
 
+#include <cstdint>
 #include <functional>
 #include <memory>
 #include <string>
-#include <utility>
 #include <vector>
 
+#include "CallstackTypes.h"
 #include "SamplingProfiler.h"
 #include "SamplingReportDataView.h"
 
 class SamplingReport {
  public:
-  explicit SamplingReport(std::shared_ptr<SamplingProfiler> a_SamplingProfiler);
+  explicit SamplingReport(std::shared_ptr<SamplingProfiler> sampling_profiler);
 
-  void FillReport();
   void UpdateReport();
-  std::shared_ptr<SamplingProfiler> GetProfiler() const { return m_Profiler; }
-  std::vector<SamplingReportDataView>& GetThreadReports() {
-    return m_ThreadReports;
-  }
-  void SetCallstackDataView(class CallStackDataView* a_DataView) {
-    m_CallstackDataView = a_DataView;
-  }
-  void OnSelectAddress(uint64_t a_Address, int32_t a_ThreadId);
-  void OnCallstackIndexChanged(size_t a_Index);
+  [[nodiscard]] std::shared_ptr<SamplingProfiler> GetProfiler() const {
+    return profiler_;
+  };
+  [[nodiscard]] std::vector<SamplingReportDataView>* GetThreadReports() {
```

Comment:
I think it is fine to keep a reference here, this is not-nullable. And pointer were required only for output parameters. Also - I think this has changed because I don't find "use pointer for output parameter" rule in the styleguide. Or is there another reason for making this return a pointer?

Reply:
I thought that we should use const reference if object won't be modified and pointer otherwise, but can't find it in style guide now, so I should revert this change.

Reply:
Yep there was a rule about this - but iirc it was only for input/output parameters. For return values it did not apply.

Reply:
From [here](https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs):
```
 Input parameters should usually be values or const references, while required (non-nullable) output and input/output parameters should usually be references. Generally, use absl::optional to represent optional inputs, and non-const pointers to represent optional outputs.
```

So I guess optional output params should be pointer, non optional refs.

Reply:
Even though I am not sure about whether this is a good idea. References are always implicit on the caller side, to it makes it more easy to mistakenly pass a ref that gets then modified.